src/main/java/com/kramekk1/medicalClinic/PatientRepository.java

```diff
@@ -44,4 +44,8 @@ public void editPatientByEmail(String email, Patient newPatient) {
         patient.setPhoneNumber(newPatient.getPhoneNumber());
     }
 
+    public void editPasswordByEmail(String email, String newPassword) {
+        Optional<Patient> patient = findPatientByEmail(email);
```

Comment:
Czy nie lepiej od razu rozpakowac optionala -> 
Patient patient = findPatientByEmail(email).orElseThrow(() -> new IllegalArgumentException("Patient with given email does not exist));

Wtedy mowimy od razu ze w przypadku jak optional bedzie pusty to ma rzucic wyjatek dzieki czemu nie musimy pozniej wszedzie tego sprawdzac a jesli nie bylo pacjenta ktoremu masz zmienic haslo to nie ma sensu dalej isc i cos probowac robic